### PR TITLE
Add hardhat-verify support for contract deployment

### PR DESCRIPTION
**Pull request by pgen0x**
This pull request adds support for seamless verification of contracts using [hardhat-verify](https://hardhat.org/hardhat-runner/plugins/nomicfoundation-hardhat-verify) It introduces the necessary mechanisms to prepare and format constructor arguments correctly for the verification process.

**Key Changes**

* Constructor arguments (token name, symbol, initial supply, etc) are now configurable using environment variables (`.env` file).
* Update `README.MD` for usage Instructions
* Update `hardhat.config.js` to add required configuration

### CODE REVIEW

File: SolidityX/package.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
     "quais-polling": "^1.0.4"
   },
   "devDependencies": {
-    "@nomicfoundation/hardhat-toolbox": "^4.0.0"
+    "@nomicfoundation/hardhat-toolbox": "^4.0.0",
+    "@nomicfoundation/hardhat-verify": "^2.0.4"
```
Comment (Juuddi) [inline SolidityX/package.json:11]: Do we need to install the `hardhat-verify` plugin? I'm pretty sure it's wrapped into the already installed `hardhat-toolbox` as you can see [here](https://hardhat.org/hardhat-runner/plugins/nomicfoundation-hardhat-toolbox).

File: Solidity/hardhat.config.js [hunk 2]
```diff
@@ -2,90 +2,166 @@
  * @type import('hardhat/config').HardhatUserConfig
  */
 
-require('@nomicfoundation/hardhat-toolbox')
-const dotenv = require('dotenv')
-dotenv.config({ path: '../.env' })
+require("@nomicfoundation/hardhat-toolbox");
+require("quai-hardhat-plugin");
```
Comment (Juuddi) [inline Solidity/hardhat.config.js:6]: the `quai-hardhat-plugin` is only relevant to the `SolidityX` folder as it downloads the SolidityX compiler. No need to import it in the config here as this only handles vanilla Solidity contracts.

File: Solidity/hardhat.config.js [hunk 3]
```diff
@@ -2,90 +2,166 @@
...
+      chainId: Number(process.env.CHAINID),
+    },
+    hydra3: {
+      url: `${process.env.HYDRA3URL}`,
+      accounts: [process.env.HYDRA3PK],
+      chainId: Number(process.env.CHAINID),
+    },
+  },
 
-	solidity: {
```
Comment (Juuddi) [inline Solidity/hardhat.config.js:59]: The Solidity contracts imported in the `Solidity` directory use version 0.8.20, not 0.8.0 like the SolidityX contracts.

File: Solidity/package.json [hunk 4]
```diff
@@ -7,6 +7,7 @@
 		"quais-polling": "^1.0.4"
 	},
 	"devDependencies": {
-		"@nomicfoundation/hardhat-toolbox": "^4.0.0"
+		"@nomicfoundation/hardhat-toolbox": "^4.0.0",
```
Comment (Juuddi) [inline Solidity/package.json:10]: See comment on `SolidityX/package.json`

File: SolidityX/hardhat.config.js [hunk 5]
```diff
@@ -2,77 +2,166 @@
  * @type import('hardhat/config').HardhatUserConfig
  */
 
-require('@nomicfoundation/hardhat-toolbox')
-require('quai-hardhat-plugin')
-const dotenv = require('dotenv')
-dotenv.config({ path: '../.env' })
+require("@nomicfoundation/hardhat-toolbox");
+require("quai-hardhat-plugin");
+require("@nomicfoundation/hardhat-verify");
```
Comment (Juuddi) [inline SolidityX/hardhat.config.js:7]: See comment on `Solidity/hardhat.config.js`

**Review by Juuddi — COMMENTED** [review]
All of these changes look good, just have a few points with the package installations that  may be duplicates.

**Comment by pgen0x** [comment]
Great, Thanks for your feedback, already updated and fixed in this commit 
[fix: Remove unnecessary dev dependencies](https://github.com/dominant-strategies/hardhat-example/pull/34/commits/6ea425661d311d440076da170de569955770300c)

**Review by Juuddi — APPROVED** [review]
Changes look good! Thx for implementing this.

**Pull request merged**